pkg/services/publicdashboards/service/query.go

```diff
@@ -260,18 +260,16 @@ func extractQueriesFromPanels(panels []interface{}, result map[int64][]*simplejs
 		for _, queryObj := range panel.Get("targets").MustArray() {
 			query := simplejson.NewFromAny(queryObj)
 
-			if hideAttr, exists := query.CheckGet("hide"); !exists || !hideAttr.MustBool() {
```

Comment:
This is the line that allow hidden queries execution. All other changes are related to tests and the new query.Service interface.